File: program/c++/learning/extends/Child.cpp
//
// Created by Administrator on 2026/7/3.
//

#include "Child.h"

#include <iostream>

Child::Child() : Parent(10) {
    std::cout << "child is &p = " << this << std::endl;
}

void Child::modify(int c) {
    //父类的私有变量不允许子类去直接访问，只能通过父类的函数去操作
    // count = c;
    set(c);
    print();
}

Child::~Child() {
    std::cout << "child 析构函数" << std::endl;
}

File: program/java/test-demo/src/main/java/com/example/test/key/Child.java
package com.example.test.key;

/**
 * @author Administrator
 * @version 1.0
 * @description: 静态方法是类，不能重写，所以多态无效
 * Parent
 * Child==father
 * @date 2026/2/24 18:56
 */
public class Child extends Parent {
    public static void get() {
        System.out.println("Child");
    }

    @Override
    public void get(String str) {
        System.out.println("Child==" + str);
    }

    public static void main(String[] args) {
        Parent parent = new Child();
        parent.get();

        parent.get("father");
    }
}
